plugins/catalog-unprocessed-entities/src/components/EntityDialog.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useState } from 'react';
+
+import Dialog from '@material-ui/core/Dialog';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import IconButton from '@material-ui/core/IconButton';
+import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
+import CloseIcon from '@material-ui/icons/Close';
+import DescriptionIcon from '@material-ui/icons/Description';
+
+import { Highlight, themes } from 'prism-react-renderer';
+
+import { UnprocessedEntity } from './../types';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    closeButton: {
+      position: 'absolute',
+      right: theme.spacing(1),
+      top: theme.spacing(1),
+      color: theme.palette.grey[500],
+    },
+    entity: {
+      overflow: 'scroll',
+      width: '100%',
+    },
+    codeBox: {
+      border: '1px solid black',
+      padding: '1em',
+    },
+  }),
+);
+
+export const EntityDialog = ({ entity }: { entity: UnprocessedEntity }) => {
+  const [open, setOpen] = useState(false);
+  const classes = useStyles();
+
+  const openDialog = () => {
+    setOpen(true);
+  };
+
+  const closeDialog = () => {
+    setOpen(false);
+  };
+
+  const dialogContent = () => {
+    return (
+      <Highlight
```

Comment:
Would it make more sense to use the `CodeSnippet` component here from `@backstage/core-components` instead? That's what the Entity Inspector does to show the raw JSON or YAML: https://github.com/backstage/backstage/blob/2d98053d5264f7b86c05b4a80024afc41fe33012/plugins/catalog-react/src/components/InspectEntityDialog/components/JsonPage.tsx#L32

Reply:
I didn't know about that, that would make more sense. WIll look into it.